### DISCUSSION TITLE
CSV file

### DISCUSSION THREAD

**Discussion by nourriahi** [idaholab/moose#24836] [Q&A General]
Hello everyone,

I am trying to use lineValuesampler to get a csv file for 80 timesteps between 0s and 3e-3s, I couldn't manage to get this, I am only obtaining for the first timestep.
```
[VectorPostprocessors]                                  
  [t_sampler]
    type = LineValueSampler
    variable = temperature
    start_point = '500 500 0'
    end_point = '500 500 100'
    num_points = 80
    sort_by = z
  []
[]

[Outputs]
  exodus = true

  [csv]
    type = CSV                                                          
    file_base = excel/additive_manufacturing_temperature_profile_output_100w
    start_time = 0.0
    end_time = 3e-3
    interval = 80
  []
[]
```
Thanks,

**Comment by GiudGiud** [comment]
Interval = 80 does not do what you think.
It triggers output every 80 time steps, not 80 times within the start/end time window

**Comment by nourriahi** [comment]
Hello,

I took your suggestions into consideration and I made the required changes, but I am still having one csv file with only one timestep (final 
 I guess), ideally I want to have a csv file with various times (interval=2)

[Outputs]
  exodus = true

  [csv]
    type = CSV                                                          # Extraction of data in the form of a CSV FILE
    file_base = excel/additive_manufacturing_temperature_profile_output_100w
    start_time = 0.0
    end_time = 3e-3
    interval = 2
    execute_on = final
  []
[]

**Reply by GiudGiud** [reply]
This is because you are using an execute_on of FINAl. This only executes once, at the end of the simulation 

You want to use an execute on of tomestep end
